.evergreen/config/functions.yml

```diff
@@ -126,6 +126,22 @@ functions:
     - command: attach.results
       params:
         file_location: "${DRIVERS_TOOLS}/results.json"
+    - command: shell.exec
+      params:
+        working_dir: src
+        script: |
+          ${PREPARE_SHELL}
+          curl -Os https://cli.codecov.io/latest/linux/codecov
```

Comment:
Since this isn't be pulled from a repository, it'd be prudent to [integrity check the binary](https://docs.codecov.com/docs/the-codecov-cli#integrity-checking-the-binary). I noted this didn't apply to the earlier GitHub Action integration (https://github.com/mongodb/mongo-php-library/commit/ae31f00855de01af72e174d949b34c1c82ca893d) since you just reference their action (`codecov/test-results-action`).

Reply:
Added the `shasum` check for the time being. I can add the gpg signature check for the checksum file as well, but with evergreen sharing hosts between runs I'm not sure if we should be messing with the keychain.